lib/util.h

```diff
@@ -319,80 +307,14 @@ extern int64_t now_ms(void);
 
 extern clock_t sclock(void);
 
-#define BUF_MMAP    (1<<1)
-
-struct buf {
-    char *s;
-    size_t len;
-    size_t alloc;
-    unsigned flags;
-};
-#define BUF_INITIALIZER { NULL, 0, 0, 0 }
-
-#define buf_new() ((struct buf *) xzmalloc(sizeof(struct buf)))
-#define buf_destroy(b) do { buf_free((b)); free((b)); } while (0)
-#define buf_ensure(b, n) do { if ((b)->alloc < (b)->len + (n)) _buf_ensure((b), (n)); } while (0)
-#define buf_putc(b, c) do { buf_ensure((b), 1); (b)->s[(b)->len++] = (c); } while (0)
-
-void _buf_ensure(struct buf *buf, size_t len);
-const char *buf_cstring(const struct buf *buf);
-const char *buf_cstringnull(const struct buf *buf);
-const char *buf_cstringnull_ifempty(const struct buf *buf);
-const char *buf_cstring_or_empty(const struct buf *buf);
-char *buf_newcstring(struct buf *buf);
-char *buf_release(struct buf *buf);
-char *buf_releasenull(struct buf *buf);
-void buf_getmap(struct buf *buf, const char **base, size_t *len);
-int buf_getline(struct buf *buf, FILE *fp);
-size_t buf_len(const struct buf *buf);
-const char *buf_base(const struct buf *buf);
-void buf_reset(struct buf *buf);
-void buf_truncate(struct buf *buf, ssize_t len);
-void buf_setcstr(struct buf *buf, const char *str);
-void buf_setmap(struct buf *buf, const char *base, size_t len);
-void buf_copy(struct buf *dst, const struct buf *src);
-void buf_append(struct buf *dst, const struct buf *src);
-void buf_appendcstr(struct buf *buf, const char *str);
-void buf_appendoverlap(struct buf *buf, const char *str);
 void buf_appendbit32(struct buf *buf, bit32 num);
 void buf_appendbit64(struct buf *buf, bit64 num);
-void buf_appendmap(struct buf *buf, const char *base, size_t len);
-void buf_cowappendmap(struct buf *buf, const char *base, unsigned int len);
-void buf_cowappendfree(struct buf *buf, char *base, unsigned int len);
-void buf_insert(struct buf *dst, unsigned int off, const struct buf *src);
-void buf_insertcstr(struct buf *buf, unsigned int off, const char *str);
-void buf_insertmap(struct buf *buf, unsigned int off, const char *base, int len);
-void buf_vprintf(struct buf *buf, const char *fmt, va_list args)
-                __attribute__((format(printf, 2, 0)));
-void buf_printf(struct buf *buf, const char *fmt, ...)
-                __attribute__((format(printf, 2, 3)));
-void buf_replace_buf(struct buf *buf, size_t offset, size_t length,
-                     const struct buf *replace);
-int buf_replace_all(struct buf *buf, const char *match,
-                    const char *replace);
-int buf_replace_char(struct buf *buf, char match, char replace);
 #ifdef ENABLE_REGEX
 int buf_replace_all_re(struct buf *buf, const regex_t *,
```

Comment:
Should the `appendbitXX` functions get refactored to `buf.h` then these seem to be the only ones left "stealing" the `buf_` prefix. We might just rename them, we have tons functions in `charset.h` et al that take a struct buf as argument for the buffer and which do not get prefixed with `buf_`.

Reply:
Also agreed.